feat(distributor): add experimental `memberlist` kvStore for ha_tracker (#10054)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

  # CLI flag: -distributor.ha-tracker.failover-timeout
  [ha_tracker_failover_timeout: <duration> | default = 30s]

>   # Backend storage to use for the ring. Please be aware that memberlist is
>   # supported by the HA tracker but its experimental.

Reviewer: ```suggestion
  # Backend storage to use for the ring. Note that memberlist
  # support by the HA tracker is experimental.
```

Reviewer: We avoid using "Please" in docs. Also, I think this sounds a bit more definitive.

Author: Thank you @tacole02 . Addressed here [94e86e8](https://github.com/grafana/mimir/pull/10054/commits/94e86e86730024d4fbf3fab2c8d47f15cd470bfc)
